src/Microsoft.TestPlatform.TestHostProvider/Hosting/DotnetTestHostManager.cs

```diff
@@ -394,14 +394,12 @@ private string GetTestHostPath(string runtimeConfigDevPath, string depsFilePath,
                 }
             }
 
-            if (string.IsNullOrEmpty(testHostPath))
-            {
-                // Try resolving testhost from output directory of test project. This is required if user has published the test project
-                // and is running tests in an isolated machine. A second scenario is self test: test platform unit tests take a project
-                // dependency on testhost (instead of nuget dependency), this drops testhost to output path.
-                testHostPath = Path.Combine(sourceDirectory, "testhost.dll");
-                EqtTrace.Verbose("DotnetTestHostManager: Assume published test project, with test host path = {0}.", testHostPath);
-            }
+            // If we are here it means it couldnt resolve testhost.dll from nuget chache.
+            // Try resolving testhost from output directory of test project. This is required if user has published the test project
+            // and is running tests in an isolated machine. A second scenario is self test: test platform unit tests take a project
+            // dependency on testhost (instead of nuget dependency), this drops testhost to output path.
+            testHostPath = Path.Combine(sourceDirectory, "testhost.dll");
```

Comment:
@smadala : We are handling the case if testhost.dll doesn't exist here https://github.com/Microsoft/vstest/blob/master/src/Microsoft.TestPlatform.TestHostProvider/Hosting/DotnetTestHostManager.cs#L243